pkg/nmstate/policy_test.go

```diff
@@ -379,6 +380,94 @@ func TestPolicyWithVlanInterface(t *testing.T) {
 	}
 }
 
+func TestPolicyWithVlanInterfaceIP(t *testing.T) {
+	testCases := []struct {
+		testNMStatePolicy *PolicyBuilder
+		expectedError     string
+		sriovInterface    string
+		vlanID            uint16
+		ipv4              string
+		ipv6              string
+	}{
+		{
+			testNMStatePolicy: buildValidPolicyTestBuilder(buildTestClientWithDummyPolicyObject()),
+			expectedError:     "",
+			sriovInterface:    "ens1",
+			vlanID:            10,
+			ipv4:              "10.10.10.10",
+			ipv6:              "2001:db8::68",
+		},
+		{
+			testNMStatePolicy: buildValidPolicyTestBuilder(buildTestClientWithDummyPolicyObject()),
+			expectedError:     "nodenetworkconfigurationpolicy 'baseInterface' cannot be empty",
+			sriovInterface:    "",
+			vlanID:            10,
+			ipv4:              "10.10.10.10",
+			ipv6:              "2001:db8::68",
+		},
+		{
+			testNMStatePolicy: buildValidPolicyTestBuilder(buildTestClientWithDummyPolicyObject()),
+			expectedError:     "invalid vlanID, allowed vlanID values are between 0-4094",
+			sriovInterface:    "ens1",
+			vlanID:            4099,
+			ipv4:              "10.10.10.10",
+			ipv6:              "2001:db8::68",
+		},
+		{
+			testNMStatePolicy: buildValidPolicyTestBuilder(buildTestClientWithDummyPolicyObject()),
+			expectedError:     "vlanInterfaceIP 'ipv4Addresses' is an invalid ipv4 address",
+			sriovInterface:    "ens1",
+			vlanID:            4099,
+			ipv4:              "",
+			ipv6:              "2001:db8::68",
+		},
+		{
+			testNMStatePolicy: buildValidPolicyTestBuilder(buildTestClientWithDummyPolicyObject()),
+			expectedError:     "vlanInterfaceIP 'ipv6Addresses' is an invalid ipv6 address",
+			sriovInterface:    "ens1",
+			vlanID:            4099,
+			ipv4:              "10.10.10.10",
+			ipv6:              "",
+		},
+	}
+	for _, testCase := range testCases {
+		testPolicy := testCase.testNMStatePolicy.WithVlanInterfaceIP(testCase.sriovInterface, testCase.vlanID,
+			testCase.ipv4, testCase.ipv6)
+		assert.Equal(t, testPolicy.errorMsg, testCase.expectedError)
```

Comment:
make sure to put the expected first, it'll make any error message if the test fails more clear
```suggestion
		assert.Equal(t, testCase.expectedError, testPolicy.errorMsg)
```

Reply:
changed